subtests/docker_cli/build/build.py

```diff
@@ -201,10 +201,14 @@ def parse_postprocess_commands(self, build_def):
 
     def basic_postprocess(self, build_def, command, parameter):
         del parameter  # not used
+        if self.config['enable_nonprintables_check']:
+            skip=['nonprintables_check']
```

Comment:
Okay, with the conditional fixed, tests pass cleanly and I think this is a reasonable workaround for the ugly situation.

Because the normal case is to disable the nonprintables check, I think the code would be much more readable and maintainable as:

    # June 2017: docker-1.13 includes ANSI codes in output, with no way to disable
    skip = ['nonprintables_check']
    if self.config['enable_nonprintables_check']:
        skip = None

This weights the default case more heavily than the exception.